### DISCUSSION TITLE
Topology does not show Client detect point's metrics

### DISCUSSION THREAD

**Discussion by nvc97** [apache/skywalking#7854] [Q&A]
Hi,
I would like to know why the Topology does not show CPM, Latency or any metric of the Client detect point (1st pic), thus, I guess that's why the links between the server point and client points are all grayed out.
On the other hand, metrics for the Tomcat point are shown just fine (2nd pic).

**Comment by wu-sheng** [comment]
CPM means you at least need 1 calls per minute. If less than that, 0 cpm shows up.
And many nodes in your topology are IP, so, they are not agent installed.

**Comment by wu-sheng** [comment]
We can't explain your local env with many details.

**Comment by nvc97** [comment]
I can assure you that there are many calls from the server to client points.
The SkyWalking version I'm using is 8.7.0, but actually I have also tested with the version 8.6.0. Although ver 8.6.0 faced the exact same issue but atleast, it can still show the Latency between those links.
Also, you said that many nodes in my topology are IP, so should I install java agent on those nodes too?

**Reply by wu-sheng** [reply]
In your screenshots, I only could say, you have traffic from some places to Tomcat, which shows in CPM.
This is not the same meaning, this Tomcat would call the IP(the line you chose) in the same load.

**Comment by nvc97** [comment]
I've just tested my same environment with the PinPoint APM and the topology can display the CPM between Tomcat and Oracle DB, Redis, Users, and other IPs like the picture below.
I wonder how different is PinPoint agent to SkyWalking java agent in terms of traffic capturing? Please enlighten me, I would really appreciate it.
![image](https://user-images.githubusercontent.com/56123257/135816839-76314c2e-bb6a-4b56-a4b2-df0a0801d214.png)

**Reply by wu-sheng** [reply]
CPM is not traffic. It is load instead. I don't know what do you mean for pinpoint, which is a different project from a different group of people.

**Reply by wu-sheng** [reply]
We don't have any relationship with pinpoint. I know them personally, that is all.
Please don't ask us to compare 2 projects, you can do that on your own. We don't want to do that from community level.

**Reply by wu-sheng** [reply]
For example, one call in 2 minutes(duration yoy chose), CPM is 0 as expected.

**Comment by wu-sheng** [comment]
If you want SkyWalkingcommunity continues on this thread, focus on us only, don't provide other project screenshot and says, a is right, b is wrong. 
We don't know what the other project is doing. We need to respect pinpoint project, rather than judging it.

If pinpoint topic keeps happening, I have no choice but to lock this discussion.

**Reply by nvc97** [reply]
ok I understand, sorry for asking that
